File: LayoutTests/http/tests/contentdispositionattachmentsandbox/resources/referer-header-stripped-frame.php
<?php
header("Content-Disposition: attachment; filename=test.html");
header("Content-Type: text/html");
?>
<!DOCTYPE html>
<head>
<?php if (isset($_GET['referrer'])) print("<meta name=\"referrer\" content=\"" . $_GET['referrer'] . "\">\n"); ?>
<style>
a {
    display: block;
    width: 100vw;
    height: 100vh;
}
</style>
</head>
<a href="echo-http-referer.php">Link to echo-http-referer.php</a>

File: LayoutTests/http/tests/contentdispositionattachmentsandbox/resources/referer-header-stripped-frame.py
#!/usr/bin/env python3

import os
import sys
from urllib.parse import parse_qs

referrer = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True).get('referrer', [''])[0]

sys.stdout.write(
    'Content-Disposition: attachment; filename=test.html\r\n'
    'Content-Type: text/html\r\n\r\n'
    '<!DOCTYPE html>\n'
    '<head>\n'
)

if referrer:
    sys.stdout.write('<meta name=\"referrer\" content=\"{}\">\n'.format(referrer))

sys.stdout.write(
    '<style>\n'
    'a {\n'
    '    display: block;\n'
    '    width: 100vw;\n'
    '    height: 100vh;\n'
    '}\n'
    '</style>\n'
    '</head>\n'
    '<a href="echo-http-referer.py">Link to echo-http-referer.py</a>\n'
)